Dashboard/app/js/lib/components/devices/AssignmentsEditView/screens/AssignDatapoints/SearchDatapoints.jsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Checkbox from 'akvo-flow/components/reusable/Checkbox';
+import AssignmentContext from '../../assignment-context';
+
+export default class SearchDatapoints extends React.Component {
+  state = {
+    selectedDatapoints: [],
+  };
+
+  onSearch = e => {
+    e.preventDefault();
+
+    const searchKey = e.target.searchDatapoints.value;
+    this.context.actions.findDatapoints(searchKey);
+  };
+
+  onSelectDatapoint = (id, checked) => {
+    const { selectedDatapoints } = this.state;
```

Comment:
maybe its more accurate to call it `selectedDataPointIds` that way we actually know whats in the array